### ISSUE TITLE
Not working with business or student account

### ISSUE THREAD

**Issue by oxosec** [lionello/onedrive-cli#4]
Please add feature to upload all type onedrive acc

**Comment by destinyb-realityi** [comment]
Any update on this?

**Comment by ShixiangWang** [comment]
+1

**Comment by dtsimon** [comment]
Bump. Is this project still active? And if so, any chance we could get a Business support? 🙏🏻🙏🏻

**Comment by lionello** [comment]
Status, since this has been open a while without a clear answer.

This can't be done incrementally. The CLI authenticates against `login.live.com` with a consumer-only client ID and the `onedrive.readwrite` scope, and talks to `api.onedrive.com/v1.0`. A work or school account can't complete that sign-in at all — it isn't a matter of a missing flag or endpoint. Business/EDU needs Azure AD authentication and `graph.microsoft.com`, which is #8 in its entirety.

So this stays open and blocked on #8, tagged `help wanted`. It is explicitly out of scope for the API-coverage work in #32, which is classic-API only.

If someone wants to take on #8, that unblocks this and I'd be glad to review it.
